### PR TITLE
 replace GMTTempFile in cases where a temporary file is being used

### PR DESCRIPTION
**Pull request by seisman**
Second PR related to #88 after #117 is merged.

### CODE REVIEW

**Comment by seisman** [comment]
Done. The PR needs to be rebased on master branch after #117 is merged.

**Comment by leouieda** [comment]
@seisman thanks for doing this! I'm about to merge #117 and then I'll merge this one. 

Please avoid including commits from other PRs because it makes it a bit harder to keep track of what is changed in each PR. It's not a big deal but might be better in the future.

File: gmt/tests/test_clib.py [hunk 1]
```diff
@@ -89,18 +89,18 @@ def test_errors_sent_to_log_file():
...
-                                        mode,
-                                        data_file.encode())
-            assert status != 0
+            with GMTTempFile() as data_file:
+                # Make a bogus module call that will fail
+                status = lib._c_call_module(lib.current_session,
+                                            'info'.encode(),
+                                            mode,
+                                            data_file.name.encode())
+                assert status != 0
```
Comment (leouieda) [inline gmt/tests/test_clib.py]: @seisman with my changes in #123, this version won't work because the temp file exists when it's created.

**Comment by leouieda** [comment]
@seisman I merged in master to this branch to finish the PR and this introduces a couple of problems (see review comments). Sorry, I should have merged this in before I did that.

File: gmt/tests/test_clib.py (hunk 1 above)
Comment (seisman) [reply gmt/tests/test_clib.py]: I just changed it back to its older version.

**Comment by seisman** [comment]
@leouieda I have rebased this branch on master, removed some commits from PR #117, and fixed the issue with `test_errors_sent_to_log_file`.

**Comment by leouieda** [comment]
@seisman merged. Sorry it took so long and thanks for putting all the work!

**Pull request merged**